### PR TITLE
Refresh user collections when a collection is saved, and don't use the current collection when editing

### LINKED ISSUE #5425: New created collection does not appear on My Collections page
STR:

1. Log in to AMO -dev
2. Go to View My Collections and click on Create a collection
3. Fill in the necessary fields and Save the new collection
4. From the user menu, select View My Collection
5. Check if the new collection is present in the list

Actual result:
The new collection is not listed in My Collections

Expected result:
The new collection is listed in My Collections

Notes:
- if you click again on Create a Collection, the form will display the values of the collection previously created
- both issues will be solved if the page is refreshed
- reproduced on AMO -dev with FF60, Win10x64

create a collection reset

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #5425 

This fixes both issues as reported at #5425.

It updates the list of user collections after one is saved, which will then allow it to appear immediately on the My Collections screen, and it also doesn't use the currently loaded collection when rendering the Create Collection form, so we don't end up with pre-populated fields.

### CODE REVIEW

File: src/amo/sagas/collections.js [hunk 1]
```diff
@@ -266,6 +266,14 @@ export function* modifyCollection(
     invariant(effectiveSlug, 'Both slug and collectionSlug cannot be empty');
     const newLocation = `/${lang}/${clientApp}/collections/${username}/${effectiveSlug}/`;
 
+    // Refresh the user's list of collections.
+    yield put(
+      fetchUserCollectionsAction({
+        errorHandlerId: errorHandler.id,
+        username,
+      }),
+    );
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply src/amo/sagas/collections.js]: So add an action which resets the state and dispatch that?

[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/sagas/collections.js]: I am not sure a new action is needed. It should be handled into the reducer (in the `case` block where a new collection is stored).

[the remark this answers is not in the document]
Comment (bobsilverberg) [reply src/amo/sagas/collections.js]: Hmm, I didn't see your last comment and just added an action, which I am now using for both create/modify and delete. I guess I can remove that code and just deal with it in the reducer instead, if you think that's better. Could you take a look at the latest commit and let me know?

**Comment by bobsilverberg** [comment]
Thanks @willdurand. This is ready for another look.

**Comment by bobsilverberg** [comment]
Thanks @willdurand

**Pull request merged**